sdk/containerregistry/azure-containers-containerregistry/TROUBLESHOOTING.md

```diff
@@ -145,7 +145,35 @@ Refer to [Troubleshoot network issues with registry](https://docs.microsoft.com/
 ```text
 AcrErrorsException: Status code 403, "{"errors":[{"code":"DENIED","message":"client with IP <> is not allowed access. Refer https://aka.ms/acr/firewall to grant access."}]}
 ```
- 
+
+## Service errors
+
+When working with `ContainerRegistryContentClient` and `ContainerRegistryAsyncContentClient` you may get an `HttpResponseException` exception with
+message containing additional information and [Docker error code](https://docs.docker.com/registry/spec/api/#errors-2).
+
+### Getting BLOB_UPLOAD_INVALID
+
+In rare cases, transient error (such as connection reset) can happen during blob upload which may lead to `ResourceNotFoundException` being thrown with message similar to
+`{"errors":[{"code":"BLOB_UPLOAD_INVALID","message":"blob upload invalid"}]}` resulting in a failed upload. In this case upload should to be restarted from the beginning.
+
+The following code snippet shows how to access detailed error information:   
+```java com.azure.containers.containerregistry.uploadBlobErrorHandling
+BinaryData configContent = BinaryData.fromObject(Collections.singletonMap("hello", "world"));
+
+try {
+    UploadRegistryBlobResult uploadResult = contentClient.uploadBlob(configContent);
+    System.out.printf("Uploaded blob: digest - '%s', size - %s\n", uploadResult.getDigest(),
+        uploadResult.getSizeInBytes());
+} catch (HttpResponseException ex) {
+    if (ex.getCause() instanceof AcrErrorsException) {
+        AcrErrorsException acrErrors = (AcrErrorsException) ex.getCause();
+        for (AcrErrorInfo info : acrErrors.getValue().getErrors()) {
```

Comment:
I don't see this type the public API [in APIView](https://apiview.dev/Assemblies/Review/7bfe65a4d43a44df831aac7f64d37fca) - do customers have access to it?